### PR TITLE
feat: add basic fauna db handler

### PR DESCRIPTION
**Pull request by Mr-Destructive**
## Description

I have added a basic handler for fauna db, it can create collections and documents.

For #7699 

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

image
select collection data

insert into collection

faunadb dashboard

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

I have added the basic faunadb handler, creating native SQL queries in faunadb might not be practical, since FQL(fauna query language) has exhaustive functions that might not work for all types of queries. Are any other workarounds possible? Happy to add more features.

Let me know if any additional requirements might be required.
Thanks.

### CODE REVIEW

File: mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+            "fauna_port": self._connection_data.get("fauna_port"),
+            "fauna_endpoint": self._connection_data.get("fauna_endpoint"),
+        }
+
+        # should have the secret
+        if (self._client_config["fauna_secret"]) is None:
+            raise Exception("FaunaDB secret is required for FaunaDB connection!")
+        # either scheme + domain + port or endpoint is required
+        # but not both
+        if self._client_config["fauna_endpoint"] is None and not (
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py:46]: Can we change this as :

```
scheme, domain, port, endpoint = (
    self._client_config["fauna_scheme"],
    self._client_config["fauna_domain"],
    self._client_config["fauna_port"],
    self._client_config["fauna_endpoint"],
)

if not (scheme and domain and port) and endpoint is None:
    raise Exception("Either scheme + domain + port or endpoint is required for FaunaDB connection!")
elif endpoint is not None and (scheme or domain or port):
    raise Exception("Either scheme + domain + port or endpoint is required for FaunaDB connection, but not both!")

```

File: mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        """Render and execute a SQL query.
+
+        Args:
+            query (ASTNode): The SQL query.
+
+        Returns:
+            Response: The query result.
+        """
+
+        if isinstance(query, Select):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py:140]: Can we just send the query to the fauna client?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py (hunk 2 above)
Comment (Mr-Destructive) [reply mindsdb/integrations/handlers/faunadb_handler/faunadb_handler.py:140]: That is a major roadblock, it accepts the [fauna query language](https://docs.fauna.com/fauna/v4/api/fql/functions/) as a [Python dialect](https://fauna.github.io/faunadb-python/4.5.0/api/query.html) and not a raw query.

**Review by ZoranPandovski — DISMISSED** [review]
Let's add initial unit tests and merge staging so the tests pass. Then we can merge this. Thanks

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
Hey @Mr-Destructive,
Typically, when we have unstructured data returned by an database/application, we try to put that data into a structured format. Is that something we can attempt to do here?
I believe you can achieve this using the `json_normalize()` method that is provided by `pandas`.

**Review by MinuraPunchihewa — COMMENTED** [review]
Hey @Mr-Destructive,
This looks good! Can you please add a couple of screenshots of the integration returning structured content? And maybe the tests passing? We can merge it then.

**Comment by Mr-Destructive** [comment]
Sure,


returnign structured data
passing tests

**Review by MinuraPunchihewa — APPROVED** [review]

**Comment by MinuraPunchihewa** [comment]
This is ready to merge.

**Pull request merged**